eXNN/InnerNeuralTopology/quantitative_indices.py

```diff
@@ -0,0 +1,63 @@
+import heapq
+import numpy as np
+
+def compute_length(diag):
+    return [d[1] - d[0] for d in diag[0]]
+
+def compute_longest_interval(lengths):
+    return max(lengths)
+
+def compute_length_mean(lengths):
+    return np.mean(lengths)
+
+def compute_length_median(lengths):
+    return np.median(lengths)
+
+def compute_length_stdev(lengths):
+    return np.std(lengths)
+
+def compute_length_sum(lengths):
+    return np.sum(lengths)
+    
+# Proportion between the longest intervals: 2/1 ratio, 3/1 ratio
+
+def compute_two_to_one_ratio(lengths):
+    return heapq.nlargest(2, lengths)[1] / lengths[0]
+
+def compute_three_to_one_ratio(lengths):
+    return heapq.nlargest(3, lengths)[2] / lengths[0]
+    
+# Compute the persistent entropy and normed persistent entropy
+def compute_entropy(lengths, length_sum):
+    return (-1) * np.sum(np.divide(lengths, length_sum)*np.log(np.divide(lengths, length_sum)))
+
+def compute_normed_entropy(entropy, length_sum):
+    return entropy / np.log(length_sum)
+    
+# Compute births
+def compute_births(diag):
+    return np.array([x[1] for x in diag[0]])
```

Comment:
There and in the next function is needed double-check - birth is start of bar in barcode, and death is it's end.

Reply:
infra comment

Reply:
For each dimension there are different barcodes, so it's true that you need to calculate metrics first each dimension separately.
About first part - okay, it's alright, there is no error

Reply:
@IvanTomilov1 
I keeped the two functions `compute_births` and `compute_deaths` and added two function `compute_birth` and `compute_death` which take two parameter (the diagram and the dimension that we want to calulte the birth and death for it)